ethology/trackers/tracker.py

```diff
@@ -0,0 +1,98 @@
+"""Trackers module."""
+
+import os
+from typing import Literal
+
+import torch
+
+from .cotracker.cotracker.utils.visualizer import (
+    Visualizer,
+    read_video_from_path,
+)
+
+
+class BaseTracker:
+    """Base class for all trackers.
+
+    Args:
+     video_source: str
+     query_points: list
+     tracker: str
+
+    """
+
+    def __init__(self, tracker_type: Literal["cotracker"]):
+        """Initialize the tracker."""
+        if tracker_type not in ["cotracker"]:
+            raise ValueError(f"Tracker type {tracker_type} not supported yet")
+
+        self.tracker = tracker_type
+
+    def track(
```

Comment:
Tried to keep the structure similar to how the function call in ultralytics looks like